Shared/Mobile-Rounds.ViewModels/Regular/Configuration/ConfigurationViewModel.cs

```diff
@@ -42,7 +45,17 @@ public ConfigurationViewModel()
 
             //now get the users metadata.
             var userInfo = await base.Api.GetAsync<UserModel>(Constants.Endpoints.Users);
-            settings.SaveValue(Constants.UserAdminKey, userInfo.IsAdministrator);
+            if (userInfo  != null)
+                settings.SaveValue(Constants.UserAdminKey, userInfo.IsAdministrator);
+        }
+
+        private async Task TestApiConnection()
+        {
+
+            //now get the users metadata.
+            var userInfo = await base.Api.GetAsync<UserModel>(Constants.Endpoints.Users);
+            if (userInfo == null)
+                throw new NullReferenceException("API Could not be contacted");
```

Comment:
I was thinking about this one the other day. We should probably add some boolean flag that **doesn't** allow the user to click "Save" if the server could not be contacted. So just adding a boolean to the `CanSave` method so that it just disables if the server is not found. That way they cannot continue to try and use the app. So maybe we do that rather than just throw the exception here. 

Think you could make this happen?